src/backend/api/meals.js

```diff
@@ -14,4 +14,80 @@ router.get("/", async (request, response) => {
   }
 });
 
+// HW part first qstn : /api/meals post adda new meal to the database
+router.post("/", async (request, response) => {
+  try {
+    const { title, description, location, when, max_reservations, price } =
+      request.body;
+    //insert to database
+    await knex("meal").insert({
+      title,
+      description,
+      location,
+      when,
+      max_reservations,
+      price,
+      created_date: new Date(),
+    });
+    response.status(201).json({ success: true });
+  } catch (error) {
+    response.status(500).json({ error: "An error occurred" });
+  }
+});
+
+// /api/meals/:id GET Returns the meal by id
+
+router.get("/:id", async (request, response) => {
+  try {
+    const mealId = request.params.id;
+    const meal = await knex("meal").where("id", mealId);
```

Comment:
`const meal = await knex("meal").where("id", mealId);` =>
`const meal = await knex("meal").where("id", mealId).first();`
Consider adding the `.first()` method, because it will retrieve the first matching record.